src/server/session.ts

```diff
@@ -1334,21 +1407,23 @@ export class Session<TMessage = string> implements EventSender {
                 goNext();
                 return;
             }
-            next.immediate("semanticCheck", () => {
-                this.semanticCheck(fileName, project);
-                if (this.changeSeq !== seq) {
-                    return;
-                }
 
-                if (this.getPreferences(fileName).disableSuggestions) {
-                    goNext();
-                    return;
-                }
-                next.immediate("suggestionCheck", () => {
-                    this.suggestionCheck(fileName, project);
-                    goNext();
+            if (ranges) {
+                return next.immediate("regionSemanticCheck", () => {
+                    const didRegionCheck = this.regionSemanticCheck(fileName, project, ranges);
+                    if (didRegionCheck) {
+                        if (this.changeSeq !== seq) {
+                            return;
+                        }
+                        next.immediate("semanticCheck", () => doSemanticCheck(fileName, project));
+                    }
+                    else {
+                        doSemanticCheck(fileName, project);
+                    }

```

Comment:
i think whether we did region check or not if "region diangostics" were expected, we should do semantic diagnostics checking on next tick